eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTests.java

```diff
@@ -0,0 +1,363 @@
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+        applications.addApplication(application);
+
+        Assert.assertNull(EurekaEntityFunctions.takeFirst(new Applications()));
+
+        Assert.assertEquals(instanceInfo,
+                EurekaEntityFunctions.takeFirst(applications));
+    }
+
+    @Test
+    public void testSelectAllIfNotNullReturnAllInstances() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+        applications.addApplication(application);
+
+        ArrayList<InstanceInfo> retval = new ArrayList<>();
+        retval.add(instanceInfo);
+        retval.add(instanceInfo);
+        Assert.assertEquals(retval,
+                EurekaEntityFunctions.selectAll(applications));
+    }
+
+    @Test
+    public void testToApplicationMapIfNotNullReturnMapOfApplication() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Mockito.when(instanceInfo.getAppName()).thenReturn("foo");
+
+        ArrayList<InstanceInfo> instanceInfos = new ArrayList<>();
+        instanceInfos.add(instanceInfo);
+
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        Assert.assertEquals(1,
+                EurekaEntityFunctions.toApplicationMap(instanceInfos).size());
+    }
+
+    @Test
+    public void
+    testToApplicationsIfNotNullReturnApplicationsFromMapoFApplication() {
+        HashMap<String, Application> hashMap = new HashMap<>();
+        hashMap.put("foo", new Application("foo"));
+        hashMap.put("bar", new Application("bar"));
+        hashMap.put("baz", new Application("baz"));
+
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, new ArrayList<>());
+        applications.addApplication(new Application("foo"));
+        applications.addApplication(new Application("bar"));
+        applications.addApplication(new Application("baz"));
+
+        Assert.assertEquals(applications.size(),
+                EurekaEntityFunctions.toApplications(hashMap).size());
+    }
+
+    @Test
+    public void testToApplicationsIfNotNullReturnApplicationsFromInstances() {
+        InstanceInfo instanceInfo1 = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo1.getId()).thenReturn("foo");
```

Comment:
change these to distinct values (app1, instance1 and so on) as technically speaking this is not valid to have instances with same ids.

Reply:
Changed